doc/source/notebooks/advanced/multioutput.ipynb

```diff
@@ -0,0 +1,747 @@
+    "    F = np.matmul(G, W)  # N x P\n",
+    "    Y = F + np.random.randn(*F.shape) * [0.2, 0.2, 0.2]\n",
+    "    \n",
+    "    return X, Y"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-06-19T10:58:46.500202Z",
+     "start_time": "2018-06-19T10:58:46.486870Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "X, Y = generate_data(N)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We create a utility function for plotting:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-06-19T10:58:48.106040Z",
+     "start_time": "2018-06-19T10:58:48.092379Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def plot_model(m, lower=-6, upper=6):\n",
+    "    pX = np.linspace(lower, upper, 100)[:, None]\n",
+    "    pY, pYv = m.predict_y(pX)\n",
+    "    if pY.ndim == 3:\n",
+    "        pY = pY[:, 0, :]\n",
+    "    plt.plot(m.X.value, m.Y.value, 'x')\n",
+    "    plt.gca().set_prop_cycle(None)\n",
+    "    plt.plot(pX, pY)\n",
+    "    for i in range(pY.shape[1]):\n",
+    "        top = pY[:, i] + 2.0 * pYv[:, i] ** 0.5\n",
+    "        bot = pY[:, i] - 2.0 * pYv[:, i] ** 0.5\n",
+    "        plt.fill_between(pX[:, 0], top, bot, alpha=0.3)\n",
+    "    plt.xlabel('X')\n",
+    "    plt.ylabel('f')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Modelling the outputs $f(x)$ directly\n",
+    "In the three examples below, we will show how to model the outputs of the model $f(x)$ directly. Mathematically, this case is equivalent to having:\n",
+    "$$\n",
+    "f(x) = I g(x),\n",
+    "$$\n",
+    "i.e. $W = I$ and $P = L$."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1. Shared Independent MOK & Shared Independent Features\n",
+    "\n",
+    "We will use the same kernel to model each of the output dimensions.\n",
+    "We will use the same inducing inputs in each of the approximations."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-06-19T10:58:58.849062Z",
+     "start_time": "2018-06-19T10:58:58.840498Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "gpf.reset_default_graph_and_session()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-06-19T10:58:59.599039Z",
+     "start_time": "2018-06-19T10:58:59.546311Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# create multioutput kernel\n",
+    "kernel = mk.SharedIndependentMok(gpf.kernels.RBF(1) + gpf.kernels.Linear(1), output_dimensionality=P) \n",
+    "# initialisation of inducing input locations (first M points of training inputs)\n",
```

Comment:
Here just using the first M is fine because the input points were randomly generated and are hence shuffled already. Would it be helpful to add a comment to that extent so people don't just end up using the first M points of a linspace()? (Yes, it *should* be obvious, but if we can easily help people not to subconsciously get it wrong then why not do it?:))